connectors/jdbc/translator-jdbc/src/test/java/org/teiid/translator/jdbc/druid/TestDruidTranslator.java

```diff
@@ -0,0 +1,149 @@
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import org.teiid.language.LanguageFactory;
+@SuppressWarnings("nls")
+public class TestDruidTranslator {
+	
+    private DruidExecutionFactory TRANSLATOR;
+    private static ExecutionContext EMPTY_CONTEXT = new FakeExecutionContextImpl();
+    private static final LanguageFactory LANG_FACTORY = new LanguageFactory();
+
+    @Before 
+    public void setup() throws Exception {
+        TRANSLATOR = new DruidExecutionFactory();
+        TRANSLATOR.setUseBindVariables(false);
+        TRANSLATOR.setDatabaseVersion(Version.DEFAULT_VERSION);
+        TRANSLATOR.start();
+    }
+
+    private String getTestVDB() {
+        return TranslationHelper.PARTS_VDB;
+    }
+
+    private String getTestBQTVDB() {
+        return TranslationHelper.BQT_VDB;
+    }
+
+    public void helpTestVisitor(String vdb, String input, String expectedOutput) throws TranslatorException {
+        TranslationHelper.helpTestVisitor(vdb, input, expectedOutput, TRANSLATOR);
+    }
+    @Test
+    public void testPushDownFuctions() throws TranslatorException {
+
+        String input = "SELECT reverse('abc') FROM BQT1.MediumA"; //$NON-NLS-1$
+        String output = "SELECT reverse('abc') FROM MediumA"; //$NON-NLS-1$
+        helpTestVisitor(getTestBQTVDB(), input, output);
+    }
+
+    @Test
+    public void testStringConversion() throws TranslatorException {
+
+        String input = "SELECT lcase(part_name) FROM Parts"; //$NON-NLS-1$
+        String output = "SELECT lower(PARTS.PART_NAME) FROM PARTS"; //$NON-NLS-1$
+        helpTestVisitor(getTestVDB(), input, output);
+
+    }
+
+    @Test
+    public void testTimeFunctions() throws TranslatorException {
+
+        String input = "select FORMATTIMESTAMP(timestampvalue, 'yyyy-MM-dd') from BQT1.MediumA"; //$NON-NLS-1$
+        String output = "SELECT TIME_FORMAT(MediumA.TimestampValue, 'yyyy-MM-dd') FROM MediumA"; //$NON-NLS-1$
+        helpTestVisitor(getTestBQTVDB(), input, output);
+
+        input = "select PARSETIMESTAMP(formattimestamp(timestampvalue, 'yyyy-MM-dd'), 'yyyy-MM-dd') from BQT1.MediumA"; //$NON-NLS-1$
+        output = "SELECT TIME_PARSE(TIME_FORMAT(MediumA.TimestampValue, 'yyyy-MM-dd'), 'yyyy-MM-dd') FROM MediumA"; //$NON-NLS-1$
+        helpTestVisitor(getTestBQTVDB(), input, output);
+
+        input = "select TIMESTAMPADD(SQL_TSI_DAY, 1, timestampvalue) from BQT1.MediumA"; //$NON-NLS-1$
+        output = "SELECT MediumA.TimestampValue + (INTERVAL '1' DAY(1)) FROM MediumA"; //$NON-NLS-1$
+        helpTestVisitor(getTestBQTVDB(), input, output);
+
+        //input = "select TIMESTAMPDIFF(SQL_TSI_DAY, NOW(), timestampvalue) from BQT1.MediumA"; //$NON-NLS-1$
+        //output = "SELECT MediumA.TimestampValue + (INTERVAL '1' DAY(1)) FROM MediumA"; //$NON-NLS-1$
+        //helpTestVisitor(getTestBQTVDB(), input, output);
+    }
+    @Test
+    public void testStringToTimestamp() throws Exception {
```

Comment:
This test fails for me locally:

Actual result: TIME_PARSE('2004-06-29 23:59:59.987', 'yyyy-MM-dd HH:mm:ss')